packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -101,15 +145,17 @@ const action: IRawAction = {
           $.app.name,
         )
       }
-      // if not all fields are used
-      const lettersErrorData: LettersApiErrorData = error.response.data
-      if (lettersErrorData.message === 'Malformed bulk create object') {
-        throw new StepError(
-          'Missing pair of field/value for letter template',
-          'Click on set up action and check that you have entered all the fields and values in the letter parameters.',
-          $.step.position,
-          $.app.name,
-        )
+      // TODO (mal): check specific fields to return
+      if (error.response.status === 400) {
+        const lettersErrorData: LettersApiFieldErrorData = error.response.data
```

Comment:
nit: can zod this

Reply:
ill need to zod this for other error catching also, will do this in another PR